axelrod/tests/unit/test_player.py

```diff
@@ -43,7 +46,21 @@ def test_noisy_play(self):
         self.assertEqual(p1.history[0], 'D')
         self.assertEqual(p2.history[0], 'D')
 
-C, D = 'C', 'D'
+
+class MockPlayer(axelrod.Player):
+    """Creates a mock player that enforces a particular next move for a given
+    player."""
+
+    def __init__(self, player, move):
+        # Need to retain history for opponents that examine opponents history
+        # Do a deep copy just to be safe
+        self.history = copy.copy(player.history)
```

Comment:
That was what confused me :)

On Fri, Jul 24, 2015 at 4:55 PM Marc Harper <notifications@github.com>
wrote:

> In axelrod/tests/unit/test_player.py
> <https://github.com/Axelrod-Python/Axelrod/pull/235#discussion_r35436260>:
>
> > @@ -43,7 +46,21 @@ def test_noisy_play(self):
> >          self.assertEqual(p1.history[0], 'D')
> >          self.assertEqual(p2.history[0], 'D')
> >
> > -C, D = 'C', 'D'
> > +
> > +class MockPlayer(axelrod.Player):
> > +    """Creates a mock player that enforces a particular next move for a given
> > +    player."""
> > +
> > +    def __init__(self, player, move):
> > +        # Need to retain history for opponents that examine opponents history
> > +        # Do a deep copy just to be safe
> > +        self.history = copy.copy(player.history)
>
> Oh yeah that should not be there! I'll fix it...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Axelrod-Python/Axelrod/pull/235/files#r35436260>.
>